massa-execution-worker/src/context.rs

```diff
@@ -113,12 +114,10 @@ impl ExecutionContext {
     pub(crate) fn new(
         final_state: Arc<RwLock<FinalState>>,
         previous_changes: StateChanges,
+        active_history: Arc<RwLock<ActiveHistory>>,
```

Comment:
Because execution _and_ speculative ledger now need to have the same history, I first tried sharing it to the speculative ledger with lifetimes but I didn't managed to do it and gave up, maybe I should give it another try. Giving a copy of the history to `ExecutionContext` when executing a new slot could be a solution as well but that seems pretty expensive to me, not sure what is the best solution here...

Reply:
@Eitu33  I made a proposal as a commit in this PR. Feel free to tell me what you think (and revert it if necessary)